runtime/runtime_impl_jetty12/src/main/java/com/google/apphosting/runtime/jetty/ee10/AppEngineWebAppContext.java

```diff
@@ -235,52 +236,59 @@ public void doStart() throws Exception {
 
     @Override
     protected void startContext() throws Exception {
-    // startWebapp is called after the web.xml metadata has been resolved, so we can
-    // clean configuration here:
-    //  - Removed deprecated filters and servlets
-    //  - Ensure known runtime filters/servlets are instantiated from this classloader
-    //  - Ensure known runtime mappings exist.
-    ServletHandler servletHandler = getServletHandler();
-    TrimmedFilters trimmedFilters =
-        new TrimmedFilters(
-            servletHandler.getFilters(),
-            servletHandler.getFilterMappings(),
-            DEPRECATED_SERVLETS_FILTERS);
-    trimmedFilters.ensure(
-        "CloudSqlConnectionCleanupFilter", JdbcMySqlConnectionCleanupFilter.class, "/*");
-
-    TrimmedServlets trimmedServlets =
-        new TrimmedServlets(
-            servletHandler.getServlets(),
-            servletHandler.getServletMappings(),
-            DEPRECATED_SERVLETS_FILTERS);
-    trimmedServlets.ensure("_ah_warmup", WarmupServlet.class, "/_ah/warmup");
-    trimmedServlets.ensure(
-        "_ah_sessioncleanup", SessionCleanupServlet.class, "/_ah/sessioncleanup");
-    trimmedServlets.ensure(
-        "_ah_queue_deferred", DeferredTaskServlet.class, "/_ah/queue/__deferred__");
-    trimmedServlets.ensure("_ah_snapshot", SnapshotServlet.class, "/_ah/snapshot");
-    trimmedServlets.ensure("_ah_default", ResourceFileServlet.class, "/");
-    trimmedServlets.ensure("default", NamedDefaultServlet.class);
-    trimmedServlets.ensure("jsp", NamedJspServlet.class);
-
-    trimmedServlets.instantiateJettyServlets();
-    trimmedFilters.instantiateJettyFilters();
-    instantiateJettyListeners();
-
-    servletHandler.setFilters(trimmedFilters.getHolders());
-    servletHandler.setFilterMappings(trimmedFilters.getMappings());
-    servletHandler.setServlets(trimmedServlets.getHolders());
-    servletHandler.setServletMappings(trimmedServlets.getMappings());
-    servletHandler.setAllowDuplicateMappings(true);
-
-    // Protect deferred task queue with constraint
-    ConstraintSecurityHandler security = (ConstraintSecurityHandler) getSecurityHandler();
-    ConstraintMapping cm = new ConstraintMapping();
-    cm.setConstraint(
-        Constraint.from("deferred_queue", Constraint.Authorization.SPECIFIC_ROLE, "admin"));
-    cm.setPathSpec("/_ah/queue/__deferred__");
-    security.addConstraintMapping(cm);
+      ServletHandler servletHandler = getServletHandler();
+      getServletHandler().addListener(new ListenerHolder() {
```

Comment:
The previous comment indicates that startContext should be called after web.xml has been resolved.   So is there a change of behaviour in ee10/jetty-12?  If so, perhaps the better fix would be to fix jetty?